### ISSUE TITLE
"ErrorCode":"mj-0004" Expected type \"emailIn\"

### ISSUE THREAD

**Issue by frcunha** [mailjet/api-documentation#31]
I'm tryng to send a e-mail with my template, code below

MailjetClient client;
		MailjetRequest request;
		MailjetResponse response;
		client = new MailjetClient(apiKey, secretKey, new ClientOptions("v3.1"));

		request = new MailjetRequest(Emailv31.resource)
				  .property(Emailv31.MESSAGES, new JSONArray()
		      .put(new JSONObject()
		        .put(Emailv31.Message.FROM, new JSONObject()
		          .put("Email", "ti@fef.edu.br")
		          .put("Name", "FEF"))
		        .put(Emailv31.Message.TO, new JSONArray()
		          .put(new JSONObject()
		            .put("Email", "passenger1@example.com")
		            .put("Name", "passenger 1")))
		        .put(Emailv31.Message.TEMPLATEID, 1483689)
		        .put(Emailv31.Message.TEMPLATELANGUAGE, true)
		        .put(Emailv31.Message.TEMPLATEERROR_REPORTING, true)
				.put(Emailv31.Message.TEMPLATEERROR_DELIVERY, true)
		        .put(Emailv31.Message.SUBJECT, "assunto")
		        .put(Emailv31.Message.VARIABLES, new JSONObject()
		        .put("titulo", "")
		        .put("saudacao", "")
		        .put("corpo", "")
		        .put("assinatura", "")
		        .put("rodape", "")
		        )
		       )
		      );

		try {
			response = client.post(request);
			System.out.println(response.getCount());
			System.out.println(response.getStatus());
			System.out.println(response.getData());
			System.out.println(response.getTotal());
		} catch (MailjetException | MailjetSocketTimeoutException e) {
			e.printStackTrace();
			
			throw new DefaultException(e.getMessage());
		}

and got this error : 

"{"ErrorIdentifier":"58c751ce-4c44-4be3-826b-81c822157765","ErrorCode":"mj-0004","StatusCode":400,"ErrorMessage":"Type mismatch. Expected type \"emailIn\".","ErrorRelatedTo":["Messages.TemplateErrorReporting"]}"

pom.xml
		<dependency>
			<groupId>com.mailjet</groupId>
			<artifactId>mailjet-client</artifactId>
			<version>4.4.0</version>
		</dependency>

Could you help me?
Sorry my poor English!

**Comment by flyerhzm** [comment]
@francoisfanuel did you solve this problem?

**Comment by frcunha** [comment]
> @francoisfanuel did you solve this problem?

Yes I did.

Wrong parameter in TEMPLATEERROR_REPORTING

**Comment by flyerhzm** [comment]
@francoisfanuel could you explain more? isn't TEMPLATEERROR_REPORTING just true or false?

**Comment by ngarnier** [comment]
@flyerhzm no, that parameter enables you to set an email address where template languate errors will be sent: https://dev.mailjet.com/email/template-language/sendapi/#templates-error-management.

You need to pass the email like this (name is optional): 
```
"TemplateErrorReporting": {
                "Email": "email@domain.com",
                "Name": "Your name"
            },
```

**Comment by gitsnaf** [comment]
It's not working for me. The documentation is quite poor concerning the values. How do I know what the value should look like?

I tried the following Java code (snippet):
`message.put(TEMPLATEID, 42)
.put(TEMPLATELANGUAGE, true)   
.put(TEMPLATEERROR_REPORTING, "email@domain.com")`
and
 `message.put(TEMPLATEID, 42)
.put(TEMPLATELANGUAGE, true)   
.put(TEMPLATEERROR_REPORTING, "{\"Email\":\"email@domain.com\"}")`
and 
 `message.put(TEMPLATEID, 42)
.put(TEMPLATELANGUAGE, true)   
.put(TEMPLATEERROR_REPORTING, "\"TemplateErrorReporting\": {\"Email\": \"email@domain.com\",\"Name\": \"Your name\"}")`

Nothing works. I always get the error:
[{"ErrorIdentifier":"43bbd862-c928-450c-9c77-db0a027a308a","ErrorCode":"mj-0004","ErrorRelatedTo":["Messages.TemplateErrorReporting"],"ErrorMessage":"Type mismatch. Expected type \"emailIn\".","StatusCode":400}]



_Update_:
I got it working:
`message.put(TEMPLATEERROR_REPORTING, new JSONObject().put(Emailv31.Message.EMAIL, "email@domain.com"))`

That did the trick.
